### PR TITLE
add Security blog post about image provenance

### PR DESCRIPTION
**Pull request by dholbach**
Thanks @pjbgf and @kingdonb for the first reviews already!

Signed-off-by: Daniel Holbach <daniel@weave.works>

### CODE REVIEW

**Comment by dholbach** [comment]
→ https://deploy-preview-762--fluxcd.netlify.app/blog/

File: content/en/blog/2022-02-14-security-image-provenance/index.md [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+> users.
+
+We are very happy that [sigstore](https://www.sigstore.dev/) exists.
+It is a Linux Foundation project backed by Google, Red Hat and Purdue
+University striving to establish a new standard for signing, verifying and
+provenance checks around the open source community.
+
+![sigstore architecture](featured-sigstore-architecture.png)
+
+The way is works is that our cosign workflow uses
```
Comment (scottrigby) [inline content/en/blog/2022-02-14-security-image-provenance/index.md:81]:
```suggestion
The way it works is that our cosign workflow uses
```

**Review by scottrigby — APPROVED** [review]
Very nice intro! One small typo, otherwise 👌 😍

File: content/en/blog/2022-02-14-security-image-provenance/index.md [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+>
+> Verification for ghcr.io/fluxcd/source-controller:v0.21.1
+> The following checks were performed on each of these signatures:>
+> - The cosign claims were validated
+> - Existence of the claims in the transparency log was verified offline
+> - Any certificates were verified against the Fulcio roots.
+> ```
+
+We are very pleased to bring this to you and encourage you to make use
+of this in your workflows to make your clusters more secure. But let's
```
Comment (pjbgf) [inline content/en/blog/2022-02-14-security-image-provenance/index.md:40]: nit:
```suggestion
of it in your workflows to make your clusters more secure. But let's
```

File: content/en/blog/2022-02-14-security-image-provenance/index.md [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+  be tampered with
+
+That's a lot of terminology and project names, but what's beautiful
+about cosign is that you can relatively easily integrate this using
+GitHub Actions (see how it was [done in
+source-controller](https://github.com/fluxcd/source-controller/pull/550/files)).
+
+## How to verify signatures
+
+If you want to do this manually as a one-off, [install the cosign
```
Comment (pjbgf) [inline content/en/blog/2022-02-14-security-image-provenance/index.md:102]: nit: 
```suggestion
If you want to do this manually as an one-off, [install the cosign
```

File: content/en/blog/2022-02-14-security-image-provenance/index.md [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+- The cosign claims were validated
+- Existence of the claims in the transparency log was verified offline
+- Any certificates were verified against the Fulcio roots.
+```
+
+Now let's take a look at how to automate this further.
+
+## Enforcing verified signatures in a cluster
+
+Luckily cosign is compatible with and supported in policy engines such
```
Comment (pjbgf) [inline content/en/blog/2022-02-14-security-image-provenance/index.md:127]: nit:
```suggestion
Luckily cosign is compatible with and supported by policy engines such
```

**Review by pjbgf — APPROVED** [review]
Awesome work putting this together @dholbach. A few nits but feel free to disregard them if they don't make sense.

File: content/en/blog/2022-02-14-security-image-provenance/index.md [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+
+The way it works is that our cosign workflow uses
+
+- [cosign](https://docs.sigstore.dev/cosign/overview) to
+  sign our release artifacts and store the signatures in an OCI
+  registry (GHCR and Docker Hub in our case)
+- OpenID Connect (OIDC) beforehand to be identified via our email
+  address
+- Fulcio, a root certification authority (CA), which issues a
+  time-stamped certificate showing the identified individual that
```
[the remark this answers is not in the document]
Comment (stefanprodan) [reply content/en/blog/2022-02-14-security-image-provenance/index.md:89]: I don't see how the signature can be linked to an individual. It's not a human that builds Flux, it's a machine account:

```
 $ COSIGN_EXPERIMENTAL=1 cosign verify ghcr.io/fluxcd/source-controller:v0.21.1

Verification for ghcr.io/fluxcd/source-controller:v0.21.1 --
The following checks were performed on each of these signatures:
  - The cosign claims were validated
  - Existence of the claims in the transparency log was verified offline
  - Any certificates were verified against the Fulcio roots.

[{"critical":{"identity":{"docker-reference":"ghcr.io/fluxcd/source-controller"},"image":{"docker-manifest-digest":"sha256:62dbdfea90baa0a08a3d3aeb11f792efd30a19a4c928cf796b0e83cddfbc2ffb"},"type":"cosign container image signature"},"optional":{"Bundle":{"SignedEntryTimestamp":"MEUCIGFZCbJdOIcqV6HGH9ljSURVpr5jCmgoch8yKMr05DGbAiEAgBjyZGV4HOc7zRco9XU0H6h/D42dybCOv+0MYJNRKIc=","Payload":{"body":"......."}},"Issuer":"https://token.actions.githubusercontent.com","Subject":"https://github.com/fluxcd/source-controller/.github/workflows/release.yml@refs/tags/v0.21.1"}}]
```

[the remark this answers is not in the document]
Comment (dholbach) [reply content/en/blog/2022-02-14-security-image-provenance/index.md:89]: ok, now I see what you're referring to - let me rephrase this :+1:

[the remark this answers is not in the document]
Comment (dholbach) [reply content/en/blog/2022-02-14-security-image-provenance/index.md]: Is this any better now?

[the remark this answers is not in the document]
Comment (stefanprodan) [reply content/en/blog/2022-02-14-security-image-provenance/index.md]: Yeap, thanks!

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @dholbach
